client/src/renderer/redux/reducers/document.js

```diff
@@ -1,5 +1,52 @@
 /** @format */
-
 import test from "./test.json";
+import { UPDATE_CARET_POSITION } from "../actions/types";
+
+const intialState = {
+  selection: {
+    start: 0,
+    end: 0,
+  },
+  nodes: translate(test),
+};
+
+function translate(nodes) {
+  let id = 0;
+  const nextId = () => id++;
+  const byId = {};
+  const addToById = node => (byId[node.id] = node);
+  const allIds = nodes.map(node => translateNode(node, nextId, addToById));
+  return { byId, allIds };
+}
+
+function translateNode(node, nextId, addToById) {
+  if (typeof node === "string") {
+    node = {
+      type: "text",
+      content: node,
+    };
+  }
```

Comment:
Please use white space to separate distinct steps in your code. In this case, the above conditional handles an edge case and therefore should be fundamentally separated.

White space is also super important for readability. Next time you feel a user interface is getting awkward or uncomfortable, try adding white space. Same deal with writing documents. Not enough white space is the number one cause of that cluttered feeling. 10/10 designers agree.

```suggestion
  }

```

Reply:
bruh

I'd argue the 7 lines of comments above the function is the number one cause of that cluttered feeling